aiida_kkr/workflows/BandStructure_WF.py

```diff
@@ -0,0 +1,443 @@
+        try:
+            outputnode_dict['nspin'] = self.ctx.BS_run.res.nspin
+        except:
+            error = "ERROR: nspin not extracted"
+            self.report(error)
+            self.ctx.successful = False
+            self.ctx.errors.append(error)
+        outputnode_dict['successful'] = self.ctx.successful
+        outputnode_dict['list_of_errors'] = self.ctx.errors
+
+        # create output node with data-provenance
+        outputnode = Dict(dict=outputnode_dict)
+        outputnode.label = 'kkr_scf_wc_results'
+        outputnode.description = ''
+
+        self.report("INFO: create Banstructure results nodes")
+        try:
+            self.report("INFO: create Bandstructure results nodes. BS calc retrieved node={}".
+                        format(self.ctx.BS_run.outputs.retrieved))
+            has_BS_run = True
+        except AttributeError as e:
+            self.report("ERROR: no Bandstructure calc retrieved node found")
+            self.report("Caught AttributeError {}".format(e))
+            has_BS_run = False
+        
+        if has_BS_run:
+            BS_retrieved = self.ctx.BS_run.outputs.retrieved
+            
+        ef = self.ctx.fermi_energy  # in eV unit
+        kpoints = self.ctx.BS_kpoints
+        outdict = {}
+        if has_BS_run:
+            ArraData = parse_BS_data(BS_retrieved, Float(ef), kpoints)
+            outdict['BS_Data'] = ArraData['BS_Data'] 
+
+        # link to the BS output nodes
+        link_nodes = outdict.copy()
+        if has_BS_run: link_nodes['BS_calc_remote'] =self.ctx.BS_run.outputs.remote_folder
+        outdict['results_wf'] = create_out_dict_node(outputnode, **link_nodes)
+
+    # create links to output nodes
+        for link_name, node in outdict.items():
+            self.out(link_name, node)
+
+        self.report("INFO: done with BS_workflow!\n")
+
+    def ev_To_Ry(value):
+        return float(value/13.6056980659)
+    
+
+@calcfunction
+def parse_BS_data(retrieved_folder, fermi_level, kpoints):
+    from masci_tools.io.common_functions import get_Ry2eV
+    from aiida.plugins import DataFactory
+    import numpy as np
+
+    eVscale = get_Ry2eV()  
+    
+    retrieved_list = retrieved_folder.list_object_names()
+    qdos_file_list = [i for i in retrieved_list if 'qdos.' in i]
+    q_vec_file = 'qvec.dat'
+
+    if q_vec_file in retrieved_list:
+        file_opened = retrieved_folder.open(q_vec_file)
+        q_vec = np.loadtxt(file_opened, skiprows=1 )
+
+    else:
+        print('{} file is not present in the retrived folder'.format(q_vec_file))
+    no_q_vec = len(q_vec[:,0])
+
+    num_dos_files = len(qdos_file_list)
+
+    with retrieved_folder.open(qdos_file_list[0]) as f:
+        total_dos = np.loadtxt(f)
+
+    for i in qdos_file_list[1:]:
+        with retrieved_folder.open(i) as f:
+            loaded_file = np.loadtxt(f)
+            total_dos[:,5:] += loaded_file[:,5:]
+    
+    ef = fermi_level.value # in eV unit
```

Comment:
this is in Ry, the conversion to eV units is only done in the next line!

Reply:
```
In [5]: ef = out['fermi_energy']

In [6]: ef
Out[6]: 0.6360325445

In [12]: ef = out['fermi_energy']

In [13]: eng_unit = out['energy_unit']

In [14]: eng_unit
Out[14]: 'eV'
```
This is from output_parameter of the parent cal. This make me a bit confused as I extract the value in the same way.

Reply:
the `energy_unit` refers to the `energy` value and the `fermi_energy_unit` gives you the unit of the `fermi_energy` which is in Ry and not in eV

Reply:
Oh, sorry. Thanks.